src/main/java/seedu/placebook/MainApp.java

```diff
@@ -112,6 +99,21 @@ private Model initModelManager(Storage storage, ReadOnlyUserPrefs userPrefs) {
             initialSchedule = new Schedule();
         }
 
+        try {
+            addressBookOptional = storage.readAddressBook();
+            if (!addressBookOptional.isPresent()) {
+                logger.info("Data file not found. Will be starting with a sample AddressBook");
+            }
+            initialData = addressBookOptional.orElseGet(SampleDataUtil::getSampleAddressBook);
+        } catch (DataConversionException e) {
+            logger.warning("Data file not in the correct format. Will be starting with an empty AddressBook");
+            initialData = new AddressBook();
+        } catch (IOException e) {
+            logger.warning("Problem while reading from the file. Will be starting with an empty AddressBook");
+            initialData = new AddressBook();
+            initialSchedule = new Schedule();
+        }
+
```

Comment:
this change in sequence might still result in valid schedule and empty addressbook? Is it that IOException is thrown when the addressbook is invalid? This might still left the problem where the addressbook does not corresponds to the schedule.

Reply:
I have not investigate the trace, please correct me if I am wrong!

Reply:
Okay I missed out the DataConversionException block, should add an additional line, initialSchedule = new Schedule() there

Reply:
With this change, we will read data from json for schedule first, then if theres any error wipe only schedule. If there is no error we will have the correct schedule.

However, if we manage to get a correct schedule, but there is error in the AddressBook, we will wipe both addressbook and schedule, so there should be no problem where theres any appointments with people that dont exist.

Reply:
Thanks for the explanation! That clear things a lot. May I ask will the loading of schedule loads those person in schedule to addressbook too?

Reply:
No it doesnt, it will read separately because they are stored in different data files.